packages/http/src/auth.ts

```diff
@@ -116,7 +116,7 @@ function makeHttpAuthRef(local: HttpAuth, reference: HttpAuth): HttpAuthRef {
     for (const flow of local.flows) {
       scopes.push(...flow.scopes.map((x) => x.value));
     }
-    return { kind: "oauth2", auth: reference, scopes: scopes };
+    return { kind: "oauth2", auth: reference, scopes: Array.from(new Set(scopes)) };
```

Comment:
This change works and produces the correct behavior. Can you update the original `scopes` declaration to be a `Set` instead of an array? We don't need to go from an array to a set to an array if we start with a set.